frame/vesting/src/lib.rs

```diff
@@ -378,45 +686,51 @@ impl<T: Config> VestingSchedule<T::AccountId> for Pallet<T> where
 		who: &T::AccountId,
 		locked: BalanceOf<T>,
 		per_block: BalanceOf<T>,
-		starting_block: T::BlockNumber
+		starting_block: T::BlockNumber,
 	) -> DispatchResult {
-		if locked.is_zero() { return Ok(()) }
-		if Vesting::<T>::contains_key(who) {
-			Err(Error::<T>::ExistingVestingSchedule)?
-		}
-		let vesting_schedule = VestingInfo {
-			locked,
-			per_block,
-			starting_block
-		};
-		Vesting::<T>::insert(who, vesting_schedule);
-		// it can't fail, but even if somehow it did, we don't really care.
-		let res = Self::update_lock(who.clone());
-		debug_assert!(res.is_ok());
+		if locked.is_zero() {
+			return Ok(());
+		}
+
+		let vesting_schedule = VestingInfo::try_new::<T>(locked, per_block, starting_block)?;
+		let mut vesting = if let Some(v) = Self::vesting(who) { v } else { BoundedVec::default() };
```

Comment:
I thought I saw a more idiomatic way to do this but can't find it now, am I missing something?

Reply:
Are you looking for

```suggestion
		let mut vesting = Self::vesting(who).unwrap_or_default();
```

?